src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Members.cs

```diff
@@ -815,17 +815,19 @@ public override void VisitParameter(IParameterSymbol symbol)
                 AddParameterRefKindIfNeeded(symbol);
                 AddCustomModifiersIfNeeded(symbol.RefCustomModifiers, leadingSpace: false, trailingSpace: true);
 
-                if (symbol.ScopedKind == ScopedKind.ScopedValue &&
-                    format.CompilerInternalOptions.IncludesOption(SymbolDisplayCompilerInternalOptions.IncludeScoped))
+                if (format.ParameterOptions.IncludesOption(SymbolDisplayParameterOptions.IncludeModifiers))
                 {
-                    AddKeyword(SyntaxKind.ScopedKeyword);
-                    AddSpace();
-                }
+                    if (symbol.ScopedKind == ScopedKind.ScopedValue)
```

Comment:
>if (symbol.ScopedKind == ScopedKind.ScopedValue) [](http://example.com/codeflow?start=20&length=48)

> We could perhaps inline ```AddParameterRefKindIfNeeded()``` at the call-site above since it's only used in that one location. That might make the sequence clearer.

I think we should seriously consider doing this and avoid introducing `ref`/`non-ref` overlapping code paths for `scoped` as we have right now. In other words, ```ScopedKind.ScopedValue``` should not affect `ref` parameters the same way as ```ScopedKind.ScopedRef``` doesn't affect value parameters. Language doesn't support these combinations and it will make the code easier to follow. BTW, I think it should be fine to do ```AddCustomModifiersIfNeeded(symbol.RefCustomModifiers ...```  only if we display `ref`, after all they are modifying the `ref` and will look very confusing without it.

Reply:
I've inlined `AddParameterRefKindIfNeeded()` but did not move `AddCustomModifiers()` into the `if (IncludeModifiers) { }` block. I agree it might be confusing to have ref custom modifiers without a leading `ref` but that seems like an unnecessary change for this PR.

Reply:
It looks like the code paths for `ref` and `ScopedKind.ScopedValue` still overlap. The primary motivation for suggesting refactoring was to avoid that.